### PR TITLE
Update for latest API

### PR DESCRIPTION
**Pull request by basta-stripe**
Some general updates for the latest API:

1. Account create no longer requires additional params
2. Account Links have new params and `type` value

### CODE REVIEW

File: server/java/src/main/java/com/stripe/sample/Server.java [hunk 1]
```diff
@@ -62,9 +61,9 @@ public static void main(String[] args) {
             String accountID = account.getId();
 
             AccountLinkCreateParams createAccountLinkParams = new AccountLinkCreateParams.Builder()
-                    .setAccount(accountID).setType("onboarding")
-                    .setFailureUrl(String.format("%s/onboard-user/refresh", origin))
-                    .setSuccessUrl(String.format("%s/success.html", origin)).build();
+                    .setAccount(accountID).setType("account_onboarding")
```
Comment (vincecima-stripe) [inline server/java/src/main/java/com/stripe/sample/Server.java:64]: Library does use enums for string values?

Comment (basta-stripe) [reply server/java/src/main/java/com/stripe/sample/Server.java:64]: It does, but it takes a String. I just work here, though.

https://github.com/stripe/stripe-java/blob/19603398bf8f29431f3c5630a2b9c6bb19c157b0/src/main/java/com/stripe/param/AccountLinkCreateParams.java#L258

Comment (auchenberg) [reply server/java/src/main/java/com/stripe/sample/Server.java:64]: Right. For consistency we should probably use the ENUM in the builder: https://stripe.dev/stripe-java/com/stripe/param/AccountLinkCreateParams.Type.html

`AccountLinkCreateParams.Type.ACCOUNT_ONBOARDING`

**Comment by auchenberg-stripe** [comment]
Overall LGTM, with one tiny nit on the enum usage.

**Pull request merged**